### PR TITLE
gperftools: Fix build on Linux

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

File: Library/Formula/google-perftools.rb [hunk 1]
```diff
@@ -16,8 +16,10 @@ class GooglePerftools < Formula
     cause "Segfault during linking"
   end
 
+  depends_on "libunwind" if OS.linux?
```
Comment (sjackman) [inline Library/Formula/google-perftools.rb]: Please change `if OS.linux?` to `unless OS.mac?`

File: Library/Formula/google-perftools.rb [hunk 2]
```diff
@@ -16,8 +16,10 @@ class GooglePerftools < Formula
...
+
   def install
-    ENV.append_to_cflags "-D_XOPEN_SOURCE"
+    ENV.append_to_cflags "-D_XOPEN_SOURCE" if OS.mac?
```
Comment (sjackman) [inline Library/Formula/google-perftools.rb]: What error message do you see without this patch chunk?

File: Library/Formula/google-perftools.rb (hunk 1 above)
Comment (dholm) [inline Library/Formula/google-perftools.rb]:
```
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/gperftools/tcmalloc.h
config.status: creating src/windows/gperftools/tcmalloc.h
config.status: creating src/config.h
config.status: executing depfiles commands
config.status: executing libtool commands
configure: error: in `/tmp/google-perftools20151005-508002-4kmgzv/gperftools-2.4':
configure: error: No frame pointers and no libunwind. The compilation will fail
See `config.log' for more details
```

File: Library/Formula/google-perftools.rb (hunk 2 above)
Comment (dholm) [inline Library/Formula/google-perftools.rb]:
```
checking for __builtin_stack_pointer()... no
checking for __builtin_expect()... yes
checking for __environ... yes
checking for __thread... yes
checking if __malloc_hook is declared volatile... no
checking if nanosleep requires any libraries... configure: error: cannot find the nanosleep function
```

File: Library/Formula/google-perftools.rb [hunk 3]
```diff
@@ -16,8 +16,10 @@ class GooglePerftools < Formula
     cause "Segfault during linking"
   end
 
+  depends_on "libunwind" unless OS.mac?
```
Comment (sjackman) [inline Library/Formula/google-perftools.rb]: Please add this comment above this line:
```ruby
# Fix error: No frame pointers and no libunwind. The compilation will fail
```

**Comment by sjackman** [comment]
I see this error
```
libtool: compile:  /gsc/btl/linuxbrew/bin/g++-5 -DHAVE_CONFIG_H -I. -I./src -I./src -isystem/gsc/btl/linuxbrew/include -Wall -Wwrite-strings -Woverloaded-virtual -Wno-sign-compare -fno-builtin-malloc -fno-builtin-free -fno-builtin-realloc -fno-builtin-calloc -fno-builtin-cfree -fno-builtin-memalign -fno-builtin-posix_memalign -fno-builtin-valloc -fno-builtin-pvalloc -Wno-unused-result -DNO_FRAME_POINTER -Os -w -pipe -march=core2 -D_XOPEN_SOURCE -c src/stacktrace.cc  -fPIC -DPIC -o src/.libs/stacktrace.o
src/stacktrace.cc:213:2: error: #error Cannot calculate stack trace: will need to write for your environment
 #error Cannot calculate stack trace: will need to write for your environment
  ^
```

File: Library/Formula/google-perftools.rb (hunk 3 above)
Comment (dholm) [inline Library/Formula/google-perftools.rb]: Done!

**Comment by dholm** [comment]
@sjackman Hmm, that error message shouldn't be printed if you have `libunwind` installed.

**Comment by sjackman** [comment]
I'm on vacation until the new year. Please bump this thread after then if you haven't heard from me.

**Comment by sjackman** [comment]
Thanks for your contribution, David.

**Pull request closed without merging**